presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -181,12 +187,41 @@ private Expression removeDynamicFilters(Expression expression, Set<String> allow
 
         private Expression removeAllDynamicFilters(Expression expression)
         {
-            DynamicFilters.ExtractResult extractResult = extractDynamicFilters(expression);
+            Expression rewrittenExpression = removeNestedDynamicFilters(expression);
+            DynamicFilters.ExtractResult extractResult = extractDynamicFilters(rewrittenExpression);
             if (extractResult.getDynamicConjuncts().isEmpty()) {
-                return expression;
+                return rewrittenExpression;
             }
             return combineConjuncts(metadata, extractResult.getStaticConjuncts());
         }
+
+        private Expression removeNestedDynamicFilters(Expression expression)
+        {
+            return ExpressionTreeRewriter.rewriteWith(new RemoveDynamicFilters(), expression);
```

Comment:
use `io.prestosql.sql.tree.ExpressionTreeRewriter#defaultRewrite`. Otherwise you will remove dynamic filters from:
```
dyn_filter1 AND dyn_filter2
```
Please add a test for that.

Reply:
When `removeNestedDynamicFilters` is called from `removeAllDynamicFilters`, we're ok to have all dynamic filters removed.
When it is called from `removeDynamicFilters`, the conjunct `dyn_filter1 AND dyn_filter2` would be broken up into the individual dynamic filters before `removeNestedDynamicFilters` is called.
In case of `filter1 AND (filter2 OR (dyn_filter1 AND dyn_filter2))` also I think we're ok to have the nested dynamic filters removed.
I believe `TestDynamicFilter::testJoinMultipleEquiJoinClauses` tests such a case.
```
Output[orderkey]
│   Layout: [orderkey:bigint]
│   Estimates: {rows: 0 (0B), cpu: 7.23M, memory: 1.55MB, network: 1.55MB}
└─ RemoteExchange[GATHER]
   │   Layout: [orderkey:bigint]
   │   Estimates: {rows: 0 (0B), cpu: 7.23M, memory: 1.55MB, network: 1.55MB}
   └─ InnerJoin[(""orderkey"" = ""orderkey_0"") AND (""custkey"" = ""partkey"")][$hashvalue, $hashvalue_8]
      │   Layout: [orderkey:bigint]
      │   Estimates: {rows: 0 (0B), cpu: 7.23M, memory: 1.55MB, network: 1.55MB}
      │   Distribution: PARTITIONED
      │   dynamicFilterAssignments = {orderkey_0 -> 264, partkey -> 265}
      ├─ RemoteExchange[REPARTITION][$hashvalue]
      │  │   Layout: [orderkey:bigint, custkey:bigint, $hashvalue:bigint]
      │  │   Estimates: {rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}
      │  └─ ScanFilterProject[table = hive:default:orders, filterPredicate = (""@$internal$dynamic_filter_function|boolean|varchar|bigint@$internal$dynamic_filter_function<t>(varchar,t):boolean""('264', ""orderkey"") AND ""@$internal$dynamic_filter_function|boolean|varchar|bigint@$internal$dynamic_filter_function<t>(varchar,t):boolean""('265', ""custkey""))]
      │         Layout: [orderkey:bigint, custkey:bigint, $hashvalue_7:bigint]
      │         Estimates: {rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}/{rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}/{rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}
      │         $hashvalue_7 := combine_hash(combine_hash(bigint '0', COALESCE(""$operator$hash_code""(""orderkey""), 0)), COALESCE(""$operator$hash_code""(""custkey""), 0))
      │         orderkey := orderkey:bigint:0:REGULAR
      │         custkey := custkey:bigint:1:REGULAR
      └─ LocalExchange[HASH][$hashvalue_8] (""orderkey_0"", ""partkey"")
         │   Layout: [orderkey_0:bigint, partkey:bigint, $hashvalue_8:bigint]
         │   Estimates: {rows: 60175 (1.55MB), cpu: 5.68M, memory: 0B, network: 1.55MB}
         └─ RemoteExchange[REPARTITION][$hashvalue_9]
            │   Layout: [orderkey_0:bigint, partkey:bigint, $hashvalue_9:bigint]
            │   Estimates: {rows: 60175 (1.55MB), cpu: 4.13M, memory: 0B, network: 1.55MB}
            └─ ScanProject[table = hive:default:lineitem]
                   Layout: [orderkey_0:bigint, partkey:bigint, $hashvalue_10:bigint]
                   Estimates: {rows: 60175 (1.55MB), cpu: 1.03M, memory: 0B, network: 0B}/{rows: 60175 (1.55MB), cpu: 2.58M, memory: 0B, network: 0B}
                   $hashvalue_10 := combine_hash(combine_hash(bigint '0', COALESCE(""$operator$hash_code""(""orderkey_0""), 0)), COALESCE(""$operator$hash_code""(""partkey""), 0))
                   partkey := partkey:bigint:1:REGULAR
                   orderkey_0 := orderkey:bigint:0:REGULAR 
```
Please let me know if I'm still missing some scenario here

Reply:
You are right, my bad